streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -210,10 +211,14 @@ public final Sensor threadLevelSensor(final String threadId,
                                           final Sensor... parents) {
         final String key = threadSensorPrefix(threadId);
         synchronized (threadLevelSensors) {
-            threadLevelSensors.putIfAbsent(key, new LinkedList<>());
             final String fullSensorName = key + SENSOR_NAME_DELIMITER + sensorName;
-            final Sensor sensor = metrics.sensor(fullSensorName, recordingLevel, parents);
-            threadLevelSensors.get(key).push(fullSensorName);
+            final Sensor sensor = Optional.ofNullable(metrics.getSensor(fullSensorName))
+                    .orElseGet(() -> {
+                        threadLevelSensors.putIfAbsent(key, new LinkedList<>());
+                        final Sensor newSensor = metrics.sensor(fullSensorName, recordingLevel, parents);
+                        threadLevelSensors.get(key).push(fullSensorName);
+                        return newSensor;
+                    });
```

Comment:
I like the use of `Optional`. I think, you could make it even simpler:
```
final Sensor sensor = Optional.ofNullable(metrics.getSensor(fullSensorName)).orElseGet(() -> {
    final Sensor newSensor = metrics.sensor(fullSensorName, recordingLevel, parents);
    threadLevelSensors.computeIfAbsent(key, ignored -> new LinkedList<>()).push(fullSensorName);
    return newSensor;
});
```
Please use the correct indentation. We use 4 spaces.

Same applies to the changes below.